File: tools/improve/tests/original/maudec/py/simple-list.py
import enum

class List(enum.Enum):
	A = 0  # 
	B = 1  # 
	C = 2  # 
	D = 3  # 


def length(a1: [List]) -> int:
	match a1:
		case []:
			return 0

		case [_, *L]:
			return length(L) + 1

File: tools/improve/tests/original/maudec/rust/simple-list.rs


#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum List {
	A,
	B,
	C,
	D,
}

fn length(a1: &[List]) -> i32 {
	match a1 {
		[] => 0,
		[_, l @ ..] => length(l) + 1,
		_ => panic!("case not covered"),
	}
}

